force_gromacs/pipeline/tests/test_gromacs_pipeline.py

```diff
@@ -25,57 +20,99 @@ def setUp(self):
 
         self.pipeline = ProbeGromacsPipeline()
 
-    def test_pipeline_getitem(self):
+    def test__len__(self):
+        self.assertEqual(4, len(self.pipeline))
+
+        new_command = Gromacs_genbox(dry_run=True)
+        self.pipeline.steps.append(('new_command', new_command))
+
+        self.assertEqual(5, len(self.pipeline))
 
-        self.assertEqual(type(self.pipeline[0]), GromacsFileTreeBuilder)
-        self.assertEqual(type(self.pipeline[1]), Gromacs_genbox)
-        self.assertEqual(type(self.pipeline[2]), Gromacs_genion)
-        self.assertEqual(type(self.pipeline[3]), GromacsTopologyWriter)
+        self.pipeline.steps.pop(0)
 
         self.assertEqual(4, len(self.pipeline))
 
-        keys = [name for name, _ in self.pipeline.steps]
+    def test_steps(self):
+
+        self.assertEqual(4, len(self.pipeline.steps))
+        self.assertIsInstance(self.pipeline[0], GromacsFileTreeBuilder)
+        self.assertIsInstance(self.pipeline[1], Gromacs_genbox)
+        self.assertIsInstance(self.pipeline[2], Gromacs_genion)
+        self.assertIsInstance(self.pipeline[3], GromacsTopologyWriter)
 
         self.assertListEqual(
-            ['file_tree', 'genbox', 'genion', 'top_file'], keys
+            ['file_tree', 'genbox', 'genion', 'top_file'],
+            [name for name, _ in self.pipeline.steps]
         )
 
         for name, command in self.pipeline.steps:
             self.assertEqual(self.pipeline.dry_run, command.dry_run)
 
     def test_named_steps(self):
 
-        self.assertIsInstance(self.pipeline.named_steps, dict)
+        self.assertEqual(4, len(self.pipeline.named_steps))
         self.assertEqual(
             ['file_tree', 'genbox', 'genion', 'top_file'],
             list(self.pipeline.named_steps.keys())
         )
 
+        self.assertIsInstance(
+            self.pipeline.named_steps['file_tree'], GromacsFileTreeBuilder
+        )
+        self.assertIsInstance(
+            self.pipeline.named_steps['genbox'], Gromacs_genbox
+        )
+        self.assertIsInstance(
+            self.pipeline.named_steps['genion'], Gromacs_genion
+        )
+        self.assertIsInstance(
+            self.pipeline.named_steps['top_file'], GromacsTopologyWriter
+        )
+
         new_command = Gromacs_genbox(dry_run=True)
         self.pipeline.append(('new_command', new_command))
         self.assertEqual(
             ['file_tree', 'genbox', 'genion', 'top_file',
              'new_command'],
             list(self.pipeline.named_steps.keys())
         )
+        self.assertIsInstance(
+            self.pipeline.named_steps['new_command'], Gromacs_genbox
+        )
 
     def test___getitem__(self):
 
         # Allow indexing by name
         self.assertEqual(
             self.pipeline[0], self.pipeline['file_tree']
         )
+        self.assertEqual(
+            self.pipeline[1], self.pipeline['genbox']
```

Comment:
Is it possible that we are actually checking for `assertIs` here?

Reply:
True, in this case we expect both variables to point to the same object.